addons/hr_timesheet/models/project_task.py

```diff
@@ -55,6 +59,17 @@ def _uom_in_days(self):
     def _compute_encode_uom_in_days(self):
         self.encode_uom_in_days = self._uom_in_days()
 
+    @api.depends('project_root_id.allow_timesheets')
+    def _compute_allow_timesheets(self):
+        for task in self:
+            task.allow_timesheets = task.project_root_id.allow_timesheets
+
+    def _search_allow_timesheets(self, operator, value):
+        query = self.env['project.project'].sudo()._search([
+            ('allow_timesheets', operator, value),
+        ])
+        return [('project_id', 'in', query)]
```

Comment:
? 🤔 
```suggestion
        return [('project_root_id', 'in', query)]
```

Reply:
That's what I did initially, but when testing it out by adding a line to a timesheet, task search caused a server error:
```
RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/home/odoo/src/odoo/16.3/odoo/odoo/http.py", line 1704, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/service/model.py", line 133, in retrying
    result = func()
  File "/home/odoo/src/odoo/16.3/odoo/odoo/http.py", line 1731, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/http.py", line 1932, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/addons/base/models/ir_http.py", line 190, in _dispatch
    result = endpoint(**request.params)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/http.py", line 716, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/home/odoo/src/odoo/16.3/odoo/addons/web/controllers/dataset.py", line 30, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/odoo/src/odoo/16.3/odoo/addons/web/controllers/dataset.py", line 26, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/api.py", line 462, in call_kw
    result = _call_kw_model(method, model, args, kwargs)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/api.py", line 435, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/models.py", line 1698, in name_search
    return self.browse(ids).sudo().name_get()
  File "/home/odoo/src/odoo/16.3/odoo/odoo/models.py", line 5380, in browse
    if not ids:
  File "/home/odoo/src/odoo/16.3/odoo/odoo/tools/query.py", line 256, in __bool__
    return bool(self.get_result_ids())
  File "/home/odoo/src/odoo/16.3/odoo/odoo/tools/query.py", line 219, in get_result_ids
    self._cr.execute(query_str, params)
  File "/home/odoo/src/odoo/16.3/odoo/odoo/sql_db.py", line 319, in execute
    res = self._obj.execute(query, params)
psycopg2.ProgrammingError: can't adapt type 'Query'

The above server error caused the following client error:
RPC_ERROR: Odoo Server Error
    at makeErrorFromResponse (http://localhost:8069/web/assets/401-cda1994/web.assets_backend.min.js:1107:163)
    at XMLHttpRequest.<anonymous> (http://localhost:8069/web/assets/401-cda1994/web.assets_backend.min.js:1111:13)
```

Reply:
Maybe I can just ignore `project_root_id`? I'd have to remove any reference to it anyway to forward port it to 16.4+.

Reply:
Guess not.
```
FAIL: TestTimesheet.test_subtask_log_timesheet
Traceback (most recent call last):
  File "/data/build/odoo/addons/hr_timesheet/tests/test_timesheet.py", line 512, in test_subtask_log_timesheet
    self.assertTrue(subtask_without_project.allow_timesheets, 'The subtask without any project set should be timesheetable because its parent task is timesheetable.')
AssertionError: False is not true : The subtask without any project set should be timesheetable because its parent task is timesheetable.
```